File: leet_029_divide_two_ints___brute.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        if divisor == 1:
            return dividend
        s = 1 if (dividend > 0 and divisor > 0) or (dividend < 0 and divisor < 0) else -1
        n = abs(dividend) // abs(divisor)
        if n > 2147483647:
            return 2147483647
        return n * s

File: leet_029_divide_two_ints___brute.js
/**
 * @param {number} dividend
 * @param {number} divisor
 * @return {number}
 */

var divide = function(dd, dv) {
    if (dv == 1)
        return dd
    sgn = (dd > 0 && dv > 0) || (dd < 0 && dv < 0) ? 1 : -1
    res = Math.floor(Math.abs(dd) / Math.abs(dv))
    if (res > 2147483647)
        return 2147483647
    return res * sgn
};
